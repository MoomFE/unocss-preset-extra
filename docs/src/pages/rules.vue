<template>
  <!-- 侧边栏 -->
  <BasicSidebar ref="sidebarRef" @setDrawerOpenBtn="(fn) => emit('setRenderNavbarLeft', fn)">
    <div class="w-60" un:flex="none">
      <div class="w-60 h-[calc(100%-64px)] fixed scrollbar" un:b-r="1 solid gray op-36">
        <n-menu v-model:value="value" :options="options" />
      </div>
    </div>
    <template #drawer>
      <n-menu v-model:value="value" :options="options" @update:value="sidebarRef.closeDrawer()" />
    </template>
  </BasicSidebar>

  <!-- 内容区域 -->
  <div class="px-6" un:p="x-6 t-5 b-10" un:flex="grow">
    <div class="text-3xl font-bold capitalize mb-1">{{ value }}</div>
    <router-view v-slot="{ Component, route }">
      <transition un:animated="~ faster" enter-active-class="animated-fade-in" leave-active-class="animated-fade-out" mode="out-in">
        <component :is="h('div', Component)" :key="route.path" />
      </transition>
    </router-view>
  </div>
</template>

<script lang="ts" setup>
  import type { MenuProps } from 'naive-ui';
  import { camelCase, upperFirst } from 'lodash';

  const emit = defineEmits(['setRenderNavbarLeft']);

  const route = useRoute();
  const router = useRouter();

  const sidebarRef = ref();

  /** 规则列表 */
  const rules = ['size', 'elevation', 'animated'];
  /** 菜单 */
  const options: MenuProps['options'] = rules.map(key => ({ label: key, key }));
  /** 当前选中的菜单项 */
  const value = ref(
    route.path.split('/').reverse()[0],
  );

  whenever(value, (key) => {
    router.push({
      name: `Rules/${upperFirst(camelCase(key))}`,
    });
  });
</script>

<route lang="yaml">
name: Rules
redirect:
  name: Rules/Size
meta:
  layout: home
  layoutContentClass: px-0!
</route>
