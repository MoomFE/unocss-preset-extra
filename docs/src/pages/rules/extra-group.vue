<template>
  <div class="text-sm">
    额外的可在分组中使用的规则, 可在
    <OuterChain href="https://github.com/unocss/unocss/tree/main/packages/transformer-variant-group" code>Variant Group</OuterChain>
    或
    <OuterChain href="https://github.com/unocss/unocss/tree/main/packages/preset-attributify" code>Attributify Mode</OuterChain>
    下使用
  </div>
  <n-divider />

  <n-h2>示例</n-h2>
  <n-card embedded>
    <CodeHighLight
      code="
        <!-- 使用之前 -->
        <div class=&quot;flex-(~ col none) justify-center&quot; />
        <div class=&quot;justify-center&quot; un:flex=&quot;~ col none&quot; />
        <!-- 使用之后 -->
        <div class=&quot;flex-(~ col none justify-center)&quot; />
        <div un:flex=&quot;~ col none justify-center&quot; />
      "
    />
  </n-card>

  <n-h2>规则</n-h2>
  <n-card>
    <CodeHighLight
      code="
        <!-- (inline-)?(flex|grid)-justify-* -->
        <div class=&quot;flex-(~ justify-start)&quot; />
        <div class=&quot;grid-(~ justify-center)&quot; />
        <div class=&quot;inline-flex-(~ justify-between)&quot; />
        <div class=&quot;inline-grid-(~ justify-end)&quot; />

        <!-- (inline-)?(flex|grid)-order-* -->
        <div class=&quot;flex-(~ order-first)&quot; />
        <div class=&quot;grid-(~ order-1)&quot; />
        <div class=&quot;inline-flex-(~ order-2)&quot; />
        <div class=&quot;inline-grid-(~ order-last)&quot; />

        <!-- (inline-)?(flex|grid)-content-* -->
        <div class=&quot;flex-(~ content-start)&quot; />
        <div class=&quot;grid-(~ content-center)&quot; />
        <div class=&quot;inline-flex-(~ content-between)&quot; />
        <div class=&quot;inline-grid-(~ content-end)&quot; />

        <!-- (inline-)?(flex|grid)-items-* -->
        <div class=&quot;flex-(~ items-start)&quot; />
        <div class=&quot;grid-(~ items-center)&quot; />
        <div class=&quot;inline-flex-(~ items-baseline)&quot; />
        <div class=&quot;inline-grid-(~ items-end)&quot; />

        <!-- (inline-)?(flex|grid)-self-* -->
        <div class=&quot;flex-(~ self-start)&quot; />
        <div class=&quot;grid-(~ self-center)&quot; />
        <div class=&quot;inline-flex-(~ self-baseline)&quot; />
        <div class=&quot;inline-grid-(~ self-end)&quot; />

        <!-- (inline-)?(flex|grid)-place-content-* -->
        <div class=&quot;flex-(~ place-content-start)&quot; />
        <div class=&quot;grid-(~ place-content-center)&quot; />
        <div class=&quot;inline-flex-(~ place-content-between)&quot; />
        <div class=&quot;inline-grid-(~ place-content-end)&quot; />

        <!-- (inline-)?(flex|grid)-place-items-* -->
        <div class=&quot;flex-(~ place-items-start)&quot; />
        <div class=&quot;grid-(~ place-items-center)&quot; />
        <div class=&quot;inline-flex-(~ place-items-stretch)&quot; />
        <div class=&quot;inline-grid-(~ place-items-end)&quot; />

        <!-- (inline-)?(flex|grid)-place-self-* -->
        <div class=&quot;flex-(~ place-self-start)&quot; />
        <div class=&quot;grid-(~ place-self-center)&quot; />
        <div class=&quot;inline-flex-(~ place-self-stretch)&quot; />
        <div class=&quot;inline-grid-(~ place-self-end)&quot; />
      "
    />
  </n-card>
</template>

<script lang="ts" setup>

</script>

<route lang="yaml">
name: Rules/ExtraGroup
</route>
