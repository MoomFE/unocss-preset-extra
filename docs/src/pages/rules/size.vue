<template>
  <div>size</div>
</template>

<script lang="ts" setup>

</script>

<route lang="yaml">
name: Rules/Size
</route>
