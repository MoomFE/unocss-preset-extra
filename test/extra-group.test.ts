import { createGenerator, presetAttributify, presetUno } from 'unocss';
import { describe, expect, test } from 'vitest';
import { createAutocomplete } from '@unocss/autocomplete';
import { isString } from '@moomfe/small-utils';
import postcss from 'postcss';
import postcssJs from 'postcss-js';
import { alignments, justifies, orders, placements } from '@unocss/preset-mini/dist/rules';
import { extraGroupRules } from '@@/src/rules/extra-group';
import { presetExtra } from '@/index';

describe('extra-group', () => {
  const generator = createGenerator({
    presets: [
      presetUno(),
      presetAttributify(),
      presetExtra(),
    ],
  });

  function addDisturbance(css: string) {
    return css.split(' ').map(s => `${s} z${s}w`).join(' ');
  }

  test('justifies', async () => {
    const staticJustifies = justifies.filter(([k]) => isString(k));
    const cssInput = addDisturbance(staticJustifies.map(([key]) => key).join(' '));
    const flexCssInput = addDisturbance(staticJustifies.map(([key]) => `flex-${key}`).join(' '));
    const gridCssInput = addDisturbance(staticJustifies.map(([key]) => `grid-${key}`).join(' '));
    const inlineFlexCssInput = addDisturbance(staticJustifies.map(([key]) => `inline-flex-${key}`).join(' '));
    const inlineGridCssInput = addDisturbance(staticJustifies.map(([key]) => `inline-grid-${key}`).join(' '));

    const { css } = await generator.generate(cssInput, { preflights: false });
    const { css: flexCss } = await generator.generate(flexCssInput, { preflights: false });
    const { css: gridCss } = await generator.generate(gridCssInput, { preflights: false });
    const { css: inlineFlexCss } = await generator.generate(inlineFlexCssInput, { preflights: false });
    const { css: inlineGridCss } = await generator.generate(inlineGridCssInput, { preflights: false });

    const cssJson = postcssJs.objectify(postcss.parse(css));
    const flexCssJson = postcssJs.objectify(postcss.parse(flexCss));
    const gridCssJson = postcssJs.objectify(postcss.parse(gridCss));
    const inlineFlexCssJson = postcssJs.objectify(postcss.parse(inlineFlexCss));
    const inlineGridCssJson = postcssJs.objectify(postcss.parse(inlineGridCss));

    Object.entries(cssJson).forEach(([key, style]) => {
      expect(style).toEqual(flexCssJson[`.flex-${key.slice(1)}`]);
      expect(style).toEqual(gridCssJson[`.grid-${key.slice(1)}`]);
      expect(style).toEqual(inlineFlexCssJson[`.inline-flex-${key.slice(1)}`]);
      expect(style).toEqual(inlineGridCssJson[`.inline-grid-${key.slice(1)}`]);
    });

    expect(`
      ${cssInput}
      ${flexCssInput}
      ${gridCssInput}
      ${inlineFlexCssInput}
      ${inlineGridCssInput}\n`.replaceAll('      ', '  ')).toMatchSnapshot();
    expect(css).toMatchSnapshot();
  });

  test('orders', async () => {
    const staticOrders = orders.filter(([k]) => isString(k));
    const cssInput = addDisturbance(staticOrders.map(([key]) => key).join(' ').concat(' order-1 order-2 order-6 order-12'));
    const flexCssInput = addDisturbance(staticOrders.map(([key]) => `flex-${key}`).join(' ').concat(' flex-order-1 flex-order-2 flex-order-6 flex-order-12'));
    const gridCssInput = addDisturbance(staticOrders.map(([key]) => `grid-${key}`).join(' ').concat(' grid-order-1 grid-order-2 grid-order-6 grid-order-12'));
    const inlineFlexCssInput = addDisturbance(staticOrders.map(([key]) => `inline-flex-${key}`).join(' ').concat(' inline-flex-order-1 inline-flex-order-2 inline-flex-order-6 inline-flex-order-12'));
    const inlineGridCssInput = addDisturbance(staticOrders.map(([key]) => `inline-grid-${key}`).join(' ').concat(' inline-grid-order-1 inline-grid-order-2 inline-grid-order-6 inline-grid-order-12'));

    const { css } = await generator.generate(cssInput, { preflights: false });
    const { css: flexCss } = await generator.generate(flexCssInput, { preflights: false });
    const { css: gridCss } = await generator.generate(gridCssInput, { preflights: false });
    const { css: inlineFlexCss } = await generator.generate(inlineFlexCssInput, { preflights: false });
    const { css: inlineGridCss } = await generator.generate(inlineGridCssInput, { preflights: false });

    const cssJson = postcssJs.objectify(postcss.parse(css));
    const flexCssJson = postcssJs.objectify(postcss.parse(flexCss));
    const gridCssJson = postcssJs.objectify(postcss.parse(gridCss));
    const inlineFlexCssJson = postcssJs.objectify(postcss.parse(inlineFlexCss));
    const inlineGridCssJson = postcssJs.objectify(postcss.parse(inlineGridCss));

    Object.entries(cssJson).forEach(([key, style]) => {
      expect(style).toEqual(flexCssJson[`.flex-${key.slice(1)}`]);
      expect(style).toEqual(gridCssJson[`.grid-${key.slice(1)}`]);
      expect(style).toEqual(inlineFlexCssJson[`.inline-flex-${key.slice(1)}`]);
      expect(style).toEqual(inlineGridCssJson[`.inline-grid-${key.slice(1)}`]);
    });

    expect(`
      ${cssInput}
      ${flexCssInput}
      ${gridCssInput}
      ${inlineFlexCssInput}
      ${inlineGridCssInput}\n`.replaceAll('      ', '  ')).toMatchSnapshot();
    expect(css).toMatchSnapshot();
  });

  test('alignments', async () => {
    const staticAlignments = alignments.filter(([k]) => isString(k));
    const cssInput = addDisturbance(staticAlignments.map(([key]) => key).join(' '));
    const flexCssInput = addDisturbance(staticAlignments.map(([key]) => `flex-${key}`).join(' '));
    const gridCssInput = addDisturbance(staticAlignments.map(([key]) => `grid-${key}`).join(' '));
    const inlineFlexCssInput = addDisturbance(staticAlignments.map(([key]) => `inline-flex-${key}`).join(' '));
    const inlineGridCssInput = addDisturbance(staticAlignments.map(([key]) => `inline-grid-${key}`).join(' '));

    const { css } = await generator.generate(cssInput, { preflights: false });
    const { css: flexCss } = await generator.generate(flexCssInput, { preflights: false });
    const { css: gridCss } = await generator.generate(gridCssInput, { preflights: false });
    const { css: inlineFlexCss } = await generator.generate(inlineFlexCssInput, { preflights: false });
    const { css: inlineGridCss } = await generator.generate(inlineGridCssInput, { preflights: false });

    const cssJson = postcssJs.objectify(postcss.parse(css));
    const flexCssJson = postcssJs.objectify(postcss.parse(flexCss));
    const gridCssJson = postcssJs.objectify(postcss.parse(gridCss));
    const inlineFlexCssJson = postcssJs.objectify(postcss.parse(inlineFlexCss));
    const inlineGridCssJson = postcssJs.objectify(postcss.parse(inlineGridCss));

    Object.entries(cssJson).forEach(([key, style]) => {
      expect(style).toEqual(flexCssJson[`.flex-${key.slice(1)}`]);
      expect(style).toEqual(gridCssJson[`.grid-${key.slice(1)}`]);
      expect(style).toEqual(inlineFlexCssJson[`.inline-flex-${key.slice(1)}`]);
      expect(style).toEqual(inlineGridCssJson[`.inline-grid-${key.slice(1)}`]);
    });

    expect(`
      ${cssInput}
      ${flexCssInput}
      ${gridCssInput}
      ${inlineFlexCssInput}
      ${inlineGridCssInput}\n`.replaceAll('      ', '  ')).toMatchSnapshot();
    expect(css).toMatchSnapshot();
  });

  test('placements', async () => {
    const staticPlacements = placements.filter(([k]) => isString(k));
    const cssInput = addDisturbance(staticPlacements.map(([key]) => key).join(' '));
    const flexCssInput = addDisturbance(staticPlacements.map(([key]) => `flex-${key}`).join(' '));
    const gridCssInput = addDisturbance(staticPlacements.map(([key]) => `grid-${key}`).join(' '));
    const inlineFlexCssInput = addDisturbance(staticPlacements.map(([key]) => `inline-flex-${key}`).join(' '));
    const inlineGridCssInput = addDisturbance(staticPlacements.map(([key]) => `inline-grid-${key}`).join(' '));

    const { css } = await generator.generate(cssInput, { preflights: false });
    const { css: flexCss } = await generator.generate(flexCssInput, { preflights: false });
    const { css: gridCss } = await generator.generate(gridCssInput, { preflights: false });
    const { css: inlineFlexCss } = await generator.generate(inlineFlexCssInput, { preflights: false });
    const { css: inlineGridCss } = await generator.generate(inlineGridCssInput, { preflights: false });

    const cssJson = postcssJs.objectify(postcss.parse(css));
    const flexCssJson = postcssJs.objectify(postcss.parse(flexCss));
    const gridCssJson = postcssJs.objectify(postcss.parse(gridCss));
    const inlineFlexCssJson = postcssJs.objectify(postcss.parse(inlineFlexCss));
    const inlineGridCssJson = postcssJs.objectify(postcss.parse(inlineGridCss));

    Object.entries(cssJson).forEach(([key, style]) => {
      expect(style).toEqual(flexCssJson[`.flex-${key.slice(1)}`]);
      expect(style).toEqual(gridCssJson[`.grid-${key.slice(1)}`]);
      expect(style).toEqual(inlineFlexCssJson[`.inline-flex-${key.slice(1)}`]);
      expect(style).toEqual(inlineGridCssJson[`.inline-grid-${key.slice(1)}`]);
    });

    expect(`
      ${cssInput}
      ${flexCssInput}
      ${gridCssInput}
      ${inlineFlexCssInput}
      ${inlineGridCssInput}\n`.replaceAll('      ', '  ')).toMatchSnapshot();
    expect(css).toMatchSnapshot();
  });

  test('autocomplete', async () => {
    const autocomplete = createAutocomplete(createGenerator({
      presets: [
        presetUno(),
        presetAttributify(),
        presetExtra(),
      ],
    }));

    // justifies
    expect(await autocomplete.suggest('flex-justify-')).toMatchSnapshot();
    expect(await autocomplete.suggest('grid-justify-')).toMatchSnapshot();
    expect(await autocomplete.suggest('inline-flex-justify-')).toMatchSnapshot();
    expect(await autocomplete.suggest('inline-grid-justify-')).toMatchSnapshot();

    // orders
    expect(await autocomplete.suggest('flex-order-')).toMatchSnapshot();
    expect(await autocomplete.suggest('grid-order-')).toMatchSnapshot();
    expect(await autocomplete.suggest('inline-flex-order-')).toMatchSnapshot();
    expect(await autocomplete.suggest('inline-grid-order-')).toMatchSnapshot();

    // alignments
    expect(await autocomplete.suggest('flex-content-')).toMatchSnapshot();
    expect(await autocomplete.suggest('flex-items-')).toMatchSnapshot();
    expect(await autocomplete.suggest('flex-self-')).toMatchSnapshot();
    expect(await autocomplete.suggest('grid-content-')).toMatchSnapshot();
    expect(await autocomplete.suggest('grid-items-')).toMatchSnapshot();
    expect(await autocomplete.suggest('grid-self-')).toMatchSnapshot();
    expect(await autocomplete.suggest('inline-flex-content-')).toMatchSnapshot();
    expect(await autocomplete.suggest('inline-flex-items-')).toMatchSnapshot();
    expect(await autocomplete.suggest('inline-flex-self-')).toMatchSnapshot();
    expect(await autocomplete.suggest('inline-grid-content-')).toMatchSnapshot();
    expect(await autocomplete.suggest('inline-grid-items-')).toMatchSnapshot();
    expect(await autocomplete.suggest('inline-grid-self-')).toMatchSnapshot();

    // placements
    expect(await autocomplete.suggest('flex-place-content-')).toMatchSnapshot();
    expect(await autocomplete.suggest('flex-place-items-')).toMatchSnapshot();
    expect(await autocomplete.suggest('flex-place-self-')).toMatchSnapshot();
    expect(await autocomplete.suggest('grid-place-content-')).toMatchSnapshot();
    expect(await autocomplete.suggest('grid-place-items-')).toMatchSnapshot();
    expect(await autocomplete.suggest('grid-place-self-')).toMatchSnapshot();
    expect(await autocomplete.suggest('inline-flex-place-content-')).toMatchSnapshot();
    expect(await autocomplete.suggest('inline-flex-place-items-')).toMatchSnapshot();
    expect(await autocomplete.suggest('inline-flex-place-self-')).toMatchSnapshot();
    expect(await autocomplete.suggest('inline-grid-place-content-')).toMatchSnapshot();
    expect(await autocomplete.suggest('inline-grid-place-items-')).toMatchSnapshot();
    expect(await autocomplete.suggest('inline-grid-place-self-')).toMatchSnapshot();
  });

  test('rules', () => {
    expect(
      extraGroupRules.map(rule => rule[0]),
    ).toMatchSnapshot();
  });
});
