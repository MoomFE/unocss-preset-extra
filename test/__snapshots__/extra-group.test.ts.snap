// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`extra-group > autocomplete 1`] = `
[
  "flex-justify-around",
  "flex-justify-between",
  "flex-justify-center",
  "flex-justify-end",
  "flex-justify-evenly",
  "flex-justify-inherit",
  "flex-justify-initial",
  "flex-justify-items-center",
  "flex-justify-items-end",
  "flex-justify-items-inherit",
  "flex-justify-items-initial",
  "flex-justify-items-revert",
  "flex-justify-items-revert-layer",
  "flex-justify-items-start",
  "flex-justify-items-stretch",
  "flex-justify-items-unset",
  "flex-justify-revert",
  "flex-justify-revert-layer",
  "flex-justify-self-auto",
  "flex-justify-self-center",
  "flex-justify-self-end",
  "flex-justify-self-inherit",
  "flex-justify-self-initial",
  "flex-justify-self-revert",
  "flex-justify-self-revert-layer",
  "flex-justify-self-start",
  "flex-justify-self-stretch",
  "flex-justify-self-unset",
  "flex-justify-start",
  "flex-justify-unset",
]
`;

exports[`extra-group > autocomplete 2`] = `
[
  "grid-justify-around",
  "grid-justify-between",
  "grid-justify-center",
  "grid-justify-end",
  "grid-justify-evenly",
  "grid-justify-inherit",
  "grid-justify-initial",
  "grid-justify-items-center",
  "grid-justify-items-end",
  "grid-justify-items-inherit",
  "grid-justify-items-initial",
  "grid-justify-items-revert",
  "grid-justify-items-revert-layer",
  "grid-justify-items-start",
  "grid-justify-items-stretch",
  "grid-justify-items-unset",
  "grid-justify-revert",
  "grid-justify-revert-layer",
  "grid-justify-self-auto",
  "grid-justify-self-center",
  "grid-justify-self-end",
  "grid-justify-self-inherit",
  "grid-justify-self-initial",
  "grid-justify-self-revert",
  "grid-justify-self-revert-layer",
  "grid-justify-self-start",
  "grid-justify-self-stretch",
  "grid-justify-self-unset",
  "grid-justify-start",
  "grid-justify-unset",
]
`;

exports[`extra-group > autocomplete 3`] = `
[
  "inline-flex-justify-around",
  "inline-flex-justify-between",
  "inline-flex-justify-center",
  "inline-flex-justify-end",
  "inline-flex-justify-evenly",
  "inline-flex-justify-inherit",
  "inline-flex-justify-initial",
  "inline-flex-justify-items-center",
  "inline-flex-justify-items-end",
  "inline-flex-justify-items-inherit",
  "inline-flex-justify-items-initial",
  "inline-flex-justify-items-revert",
  "inline-flex-justify-items-revert-layer",
  "inline-flex-justify-items-start",
  "inline-flex-justify-items-stretch",
  "inline-flex-justify-items-unset",
  "inline-flex-justify-revert",
  "inline-flex-justify-revert-layer",
  "inline-flex-justify-self-auto",
  "inline-flex-justify-self-center",
  "inline-flex-justify-self-end",
  "inline-flex-justify-self-inherit",
  "inline-flex-justify-self-initial",
  "inline-flex-justify-self-revert",
  "inline-flex-justify-self-revert-layer",
  "inline-flex-justify-self-start",
  "inline-flex-justify-self-stretch",
  "inline-flex-justify-self-unset",
  "inline-flex-justify-start",
  "inline-flex-justify-unset",
]
`;

exports[`extra-group > autocomplete 4`] = `
[
  "inline-grid-justify-around",
  "inline-grid-justify-between",
  "inline-grid-justify-center",
  "inline-grid-justify-end",
  "inline-grid-justify-evenly",
  "inline-grid-justify-inherit",
  "inline-grid-justify-initial",
  "inline-grid-justify-items-center",
  "inline-grid-justify-items-end",
  "inline-grid-justify-items-inherit",
  "inline-grid-justify-items-initial",
  "inline-grid-justify-items-revert",
  "inline-grid-justify-items-revert-layer",
  "inline-grid-justify-items-start",
  "inline-grid-justify-items-stretch",
  "inline-grid-justify-items-unset",
  "inline-grid-justify-revert",
  "inline-grid-justify-revert-layer",
  "inline-grid-justify-self-auto",
  "inline-grid-justify-self-center",
  "inline-grid-justify-self-end",
  "inline-grid-justify-self-inherit",
  "inline-grid-justify-self-initial",
  "inline-grid-justify-self-revert",
  "inline-grid-justify-self-revert-layer",
  "inline-grid-justify-self-start",
  "inline-grid-justify-self-stretch",
  "inline-grid-justify-self-unset",
  "inline-grid-justify-start",
  "inline-grid-justify-unset",
]
`;

exports[`extra-group > autocomplete 5`] = `
[
  "flex-order-first",
  "flex-order-last",
  "flex-order-none",
]
`;

exports[`extra-group > autocomplete 6`] = `
[
  "grid-order-first",
  "grid-order-last",
  "grid-order-none",
]
`;

exports[`extra-group > autocomplete 7`] = `
[
  "inline-flex-order-first",
  "inline-flex-order-last",
  "inline-flex-order-none",
]
`;

exports[`extra-group > autocomplete 8`] = `
[
  "inline-grid-order-first",
  "inline-grid-order-last",
  "inline-grid-order-none",
]
`;

exports[`extra-group > autocomplete 9`] = `
[
  "flex-content-around",
  "flex-content-between",
  "flex-content-center",
  "flex-content-end",
  "flex-content-evenly",
  "flex-content-inherit",
  "flex-content-initial",
  "flex-content-revert",
  "flex-content-revert-layer",
  "flex-content-start",
  "flex-content-unset",
]
`;

exports[`extra-group > autocomplete 10`] = `
[
  "flex-items-baseline",
  "flex-items-center",
  "flex-items-end",
  "flex-items-inherit",
  "flex-items-initial",
  "flex-items-revert",
  "flex-items-revert-layer",
  "flex-items-start",
  "flex-items-stretch",
  "flex-items-unset",
]
`;

exports[`extra-group > autocomplete 11`] = `
[
  "flex-self-auto",
  "flex-self-baseline",
  "flex-self-center",
  "flex-self-end",
  "flex-self-inherit",
  "flex-self-initial",
  "flex-self-revert",
  "flex-self-revert-layer",
  "flex-self-start",
  "flex-self-stretch",
  "flex-self-unset",
]
`;

exports[`extra-group > autocomplete 12`] = `
[
  "grid-content-around",
  "grid-content-between",
  "grid-content-center",
  "grid-content-end",
  "grid-content-evenly",
  "grid-content-inherit",
  "grid-content-initial",
  "grid-content-revert",
  "grid-content-revert-layer",
  "grid-content-start",
  "grid-content-unset",
]
`;

exports[`extra-group > autocomplete 13`] = `
[
  "grid-items-baseline",
  "grid-items-center",
  "grid-items-end",
  "grid-items-inherit",
  "grid-items-initial",
  "grid-items-revert",
  "grid-items-revert-layer",
  "grid-items-start",
  "grid-items-stretch",
  "grid-items-unset",
]
`;

exports[`extra-group > autocomplete 14`] = `
[
  "grid-self-auto",
  "grid-self-baseline",
  "grid-self-center",
  "grid-self-end",
  "grid-self-inherit",
  "grid-self-initial",
  "grid-self-revert",
  "grid-self-revert-layer",
  "grid-self-start",
  "grid-self-stretch",
  "grid-self-unset",
]
`;

exports[`extra-group > autocomplete 15`] = `
[
  "inline-flex-content-around",
  "inline-flex-content-between",
  "inline-flex-content-center",
  "inline-flex-content-end",
  "inline-flex-content-evenly",
  "inline-flex-content-inherit",
  "inline-flex-content-initial",
  "inline-flex-content-revert",
  "inline-flex-content-revert-layer",
  "inline-flex-content-start",
  "inline-flex-content-unset",
]
`;

exports[`extra-group > autocomplete 16`] = `
[
  "inline-flex-items-baseline",
  "inline-flex-items-center",
  "inline-flex-items-end",
  "inline-flex-items-inherit",
  "inline-flex-items-initial",
  "inline-flex-items-revert",
  "inline-flex-items-revert-layer",
  "inline-flex-items-start",
  "inline-flex-items-stretch",
  "inline-flex-items-unset",
]
`;

exports[`extra-group > autocomplete 17`] = `
[
  "inline-flex-self-auto",
  "inline-flex-self-baseline",
  "inline-flex-self-center",
  "inline-flex-self-end",
  "inline-flex-self-inherit",
  "inline-flex-self-initial",
  "inline-flex-self-revert",
  "inline-flex-self-revert-layer",
  "inline-flex-self-start",
  "inline-flex-self-stretch",
  "inline-flex-self-unset",
]
`;

exports[`extra-group > autocomplete 18`] = `
[
  "inline-grid-content-around",
  "inline-grid-content-between",
  "inline-grid-content-center",
  "inline-grid-content-end",
  "inline-grid-content-evenly",
  "inline-grid-content-inherit",
  "inline-grid-content-initial",
  "inline-grid-content-revert",
  "inline-grid-content-revert-layer",
  "inline-grid-content-start",
  "inline-grid-content-unset",
]
`;

exports[`extra-group > autocomplete 19`] = `
[
  "inline-grid-items-baseline",
  "inline-grid-items-center",
  "inline-grid-items-end",
  "inline-grid-items-inherit",
  "inline-grid-items-initial",
  "inline-grid-items-revert",
  "inline-grid-items-revert-layer",
  "inline-grid-items-start",
  "inline-grid-items-stretch",
  "inline-grid-items-unset",
]
`;

exports[`extra-group > autocomplete 20`] = `
[
  "inline-grid-self-auto",
  "inline-grid-self-baseline",
  "inline-grid-self-center",
  "inline-grid-self-end",
  "inline-grid-self-inherit",
  "inline-grid-self-initial",
  "inline-grid-self-revert",
  "inline-grid-self-revert-layer",
  "inline-grid-self-start",
  "inline-grid-self-stretch",
  "inline-grid-self-unset",
]
`;

exports[`extra-group > autocomplete 21`] = `
[
  "flex-place-content-around",
  "flex-place-content-between",
  "flex-place-content-center",
  "flex-place-content-end",
  "flex-place-content-evenly",
  "flex-place-content-inherit",
  "flex-place-content-initial",
  "flex-place-content-revert",
  "flex-place-content-revert-layer",
  "flex-place-content-start",
  "flex-place-content-stretch",
  "flex-place-content-unset",
]
`;

exports[`extra-group > autocomplete 22`] = `
[
  "flex-place-items-center",
  "flex-place-items-end",
  "flex-place-items-inherit",
  "flex-place-items-initial",
  "flex-place-items-revert",
  "flex-place-items-revert-layer",
  "flex-place-items-start",
  "flex-place-items-stretch",
  "flex-place-items-unset",
]
`;

exports[`extra-group > autocomplete 23`] = `
[
  "flex-place-self-auto",
  "flex-place-self-center",
  "flex-place-self-end",
  "flex-place-self-inherit",
  "flex-place-self-initial",
  "flex-place-self-revert",
  "flex-place-self-revert-layer",
  "flex-place-self-start",
  "flex-place-self-stretch",
  "flex-place-self-unset",
]
`;

exports[`extra-group > autocomplete 24`] = `
[
  "grid-place-content-around",
  "grid-place-content-between",
  "grid-place-content-center",
  "grid-place-content-end",
  "grid-place-content-evenly",
  "grid-place-content-inherit",
  "grid-place-content-initial",
  "grid-place-content-revert",
  "grid-place-content-revert-layer",
  "grid-place-content-start",
  "grid-place-content-stretch",
  "grid-place-content-unset",
]
`;

exports[`extra-group > autocomplete 25`] = `
[
  "grid-place-items-center",
  "grid-place-items-end",
  "grid-place-items-inherit",
  "grid-place-items-initial",
  "grid-place-items-revert",
  "grid-place-items-revert-layer",
  "grid-place-items-start",
  "grid-place-items-stretch",
  "grid-place-items-unset",
]
`;

exports[`extra-group > autocomplete 26`] = `
[
  "grid-place-self-auto",
  "grid-place-self-center",
  "grid-place-self-end",
  "grid-place-self-inherit",
  "grid-place-self-initial",
  "grid-place-self-revert",
  "grid-place-self-revert-layer",
  "grid-place-self-start",
  "grid-place-self-stretch",
  "grid-place-self-unset",
]
`;

exports[`extra-group > autocomplete 27`] = `
[
  "inline-flex-place-content-around",
  "inline-flex-place-content-between",
  "inline-flex-place-content-center",
  "inline-flex-place-content-end",
  "inline-flex-place-content-evenly",
  "inline-flex-place-content-inherit",
  "inline-flex-place-content-initial",
  "inline-flex-place-content-revert",
  "inline-flex-place-content-revert-layer",
  "inline-flex-place-content-start",
  "inline-flex-place-content-stretch",
  "inline-flex-place-content-unset",
]
`;

exports[`extra-group > autocomplete 28`] = `
[
  "inline-flex-place-items-center",
  "inline-flex-place-items-end",
  "inline-flex-place-items-inherit",
  "inline-flex-place-items-initial",
  "inline-flex-place-items-revert",
  "inline-flex-place-items-revert-layer",
  "inline-flex-place-items-start",
  "inline-flex-place-items-stretch",
  "inline-flex-place-items-unset",
]
`;

exports[`extra-group > autocomplete 29`] = `
[
  "inline-flex-place-self-auto",
  "inline-flex-place-self-center",
  "inline-flex-place-self-end",
  "inline-flex-place-self-inherit",
  "inline-flex-place-self-initial",
  "inline-flex-place-self-revert",
  "inline-flex-place-self-revert-layer",
  "inline-flex-place-self-start",
  "inline-flex-place-self-stretch",
  "inline-flex-place-self-unset",
]
`;

exports[`extra-group > autocomplete 30`] = `
[
  "inline-grid-place-content-around",
  "inline-grid-place-content-between",
  "inline-grid-place-content-center",
  "inline-grid-place-content-end",
  "inline-grid-place-content-evenly",
  "inline-grid-place-content-inherit",
  "inline-grid-place-content-initial",
  "inline-grid-place-content-revert",
  "inline-grid-place-content-revert-layer",
  "inline-grid-place-content-start",
  "inline-grid-place-content-stretch",
  "inline-grid-place-content-unset",
]
`;

exports[`extra-group > autocomplete 31`] = `
[
  "inline-grid-place-items-center",
  "inline-grid-place-items-end",
  "inline-grid-place-items-inherit",
  "inline-grid-place-items-initial",
  "inline-grid-place-items-revert",
  "inline-grid-place-items-revert-layer",
  "inline-grid-place-items-start",
  "inline-grid-place-items-stretch",
  "inline-grid-place-items-unset",
]
`;

exports[`extra-group > autocomplete 32`] = `
[
  "inline-grid-place-self-auto",
  "inline-grid-place-self-center",
  "inline-grid-place-self-end",
  "inline-grid-place-self-inherit",
  "inline-grid-place-self-initial",
  "inline-grid-place-self-revert",
  "inline-grid-place-self-revert-layer",
  "inline-grid-place-self-start",
  "inline-grid-place-self-stretch",
  "inline-grid-place-self-unset",
]
`;
